### ISSUE TITLE
Failing tests

### ISSUE THREAD

**Issue by sanjayankur31** [tebeka/fastavro#147]
Hi,

When I attempt to build fastavro on the Fedora builders, the build for i686 fails with two failed tests:

```
=================================== FAILURES ===================================
_____________________________ test_complex_schema ______________________________
Traceback (most recent call last):
  File "/builddir/build/BUILD/fastavro-0.17.3/tests/test_complex.py", line 90, in test_complex_schema
    assert (data1 == data2)
AssertionError: assert {'array_bytes...b', 'c'], ...} == {'array_bytes_...b', 'c'], ...}
  Omitting 4 identical items, use -vv to show
  Differing items:
  {'multi_union_time': datetime.datetime(2018, 1, 22, 20, 57, 1, 484854)} != {'multi_union_time': datetime.datetime(2018, 1, 22, 20, 57, 1, 484853)}
  Use -v to get the full diff
______________________________ test_logical_types ______________________________
Traceback (most recent call last):
  File "/builddir/build/BUILD/fastavro-0.17.3/tests/test_logical_types.py", line 74, in test_logical_types
    assert (data1['timestamp-micros'] == data2['timestamp-micros'])
AssertionError: assert datetime.datetime(2018, 1, 22, 20, 57, 1, 713539) == datetime.datetime(2018, 1, 22, 20, 57, 1, 713538)
=============== 2 failed, 100 passed, 1 skipped in 0.47 seconds ================
```

There isn't any issue on any of the [other architectures](https://koji.fedoraproject.org/koji/taskinfo?taskID=24377414). The complete build log is also attached. Any hints on this?
[build-log.txt](https://github.com/tebeka/fastavro/files/1653597/build-log.txt)

**Comment by scottbelden** [comment]
That's pretty strange. What version of python is this running with? We had to skip one of the datetime tests on Windows with python 3.6 due to a python bug (https://bugs.python.org/issue29097). I wonder if this is triggering another platform specific bug.

Did this just start failing on 0.17.3? If so it seems like it might be a fluke as the only changes between 0.17.2 and 0.17.3 were to help with publishing wheels on windows (https://github.com/tebeka/fastavro/compare/0.17.2...0.17.3)

**Comment by sanjayankur31** [comment]
It does seem to be with python3, yes:

```
============================= test session starts ==============================
platform linux -- Python 3.6.4, pytest-3.2.3, py-1.5.2, pluggy-0.4.0
rootdir: /builddir/build/BUILD/fastavro-0.17.3, inifile: pytest.ini
```

**Comment by scottbelden** [comment]
Do you run these builds often? Is it consistently failing?

**Comment by sanjayankur31** [comment]
I only ran the build once before - to import it into the Fedora repositories. I've run more now. It doesn't seem to fail each time. It passed [here](https://koji.fedoraproject.org/koji/taskinfo?taskID=24403237) for example when I tried to build it only for i686. However, when I attempt to build it for all supported arches, it fails consistently. [Here](https://koji.fedoraproject.org/koji/taskinfo?taskID=24403322), and [here](https://koji.fedoraproject.org/koji/taskinfo?taskID=24403218), for example. So, it could be a weird timing issue---these builds run in parallel, and it only fails on i686. 

Unfortunately, while I can run arch specific test (scratch) builds, I cannot run arch specific builds for the repositories, so I've disabled the tests for the time being.
